pymeasure/instruments/instrument.py

```diff
@@ -405,7 +465,10 @@ def fset(self,
                     'Values of type `{}` are not allowed '
                     'for Instrument.control'.format(type(values))
                 )
-            self.write(command_process(set_command) % value)
+            if self.read_set_reply:
+                self.ask(command_process(set_command) % value)
+            else:
+                self.write(command_process(set_command) % value)
```

Comment:
What do you think about this implementation I proposed in #617 ?
(It's just a single commit, easy to remove again)

It breaks some instruments, which use `Instrument.control`, but are not based on `Instrument` such that they miss a `self.read_set_reply`. 

As a fix, I could create a channel base class, which has this attribute and which could be later improved to contain dynamic properties, see #616 .

Reply:
I see your need to implement this, and we should have some way of checking an acknowledgement. However, I'd love to do this later, to keep the scope of this PR limited. So if it does not make some other part of this PR's work difficult to implement, please let us do that in a follow-up work (e.g. as part of resolving #617).

> It breaks some instruments, which use Instrument.control, but are not based on Instrument such that they miss a self.read_set_reply.

Classes using `control` and friends without inheriting from `Instrument` is an antipattern and should be removed (and future use discouraged). As you know it also breaks dynamic properties. From what I have seen so far, it's only channel implementations that use it. I think it makes most sense to add a `ChannelBase` (or so) class and extract the needed parts in common with `Instrument` to a common superclass for both `Instrument` and `ChannelBase`. This should also fix the dynamic property problem.